Test/Framework/Input/GamePadTest.cs

```diff
@@ -0,0 +1,136 @@
+        {
+            Assert.IsInstanceOf<GamePadState>(GamePad.GetState(i));
+        }
+
+        [Test]
+        public void GetState_ShouldNotReturnNull_WhenGivenPlayerIndex_GamePadDeadZone(
+            [ValueSource("GetPlayerIndexes")] PlayerIndex pi, // generate tests for each combination of PlayerIndex and GamePadDeadZone enum values
+            [ValueSource("GetGamePadDeadZoneIndexes")] GamePadDeadZone gpdz
+            )
+        {
+            Assert.IsNotNull(GamePad.GetState(pi, gpdz));
+        }
+
+        [Test]
+        public void GetState_ShouldReturnGamePadState_WhenGivenPlayerIndex_GamePadDeadZone(
+            [ValueSource("GetPlayerIndexes")] PlayerIndex pi, // generate tests for each combination of PlayerIndex and GamePadDeadZone enum values
+            [ValueSource("GetGamePadDeadZoneIndexes")] GamePadDeadZone gpdz
+            )
+        {
+            Assert.IsInstanceOf<GamePadState>(GamePad.GetState(pi, gpdz));
+        }
+
+        [Test]
+        public void GetState_ShouldNotReturnNull_WhenGivenInt_GamePadDeadZone(
+            [Range(-1, 4)] int i, // generate tests for each combination of the index range and GamePadDeadZone enum values
+            [ValueSource("GetGamePadDeadZoneIndexes")] GamePadDeadZone gpdz
+            )
+        {
+            Assert.IsNotNull(GamePad.GetState(i, gpdz));
+        }
+
+        [Test]
+        public void GetState_ShouldReturnGamePadState_WhenGivenInt_GamePadeDeadZone(
+            [Range(-1, 4)] int i, // generate tests for each combination of the index range and GamePadDeadZone enum values
+            [ValueSource("GetGamePadDeadZoneIndexes")] GamePadDeadZone gpdz
+            )
+        {
+            Assert.IsInstanceOf<GamePadState>(GamePad.GetState(i, gpdz));
+        }
+
+        [Test]
+        public void SetVibration_ShouldNotReturnNull_WhenGivenPlayerIndex_Float_Float(
+            [ValueSource("GetPlayerIndexes")] PlayerIndex pi, // generate tests for each combination of the PlayerIndex enum values and both float ranges
+            [Range(0.0f, 1.0f, 0.25f)] float lm,
+            [Range(0.0f, 1.0f, 0.25f)] float rm
+            )
+        {
+            Assert.IsNotNull(GamePad.SetVibration(pi, lm, rm));
+        }
+
+        [Test]
+        public void SetVibration_ShouldReturnBool_WhenGivenPlayerIndex_Float_Float(
+            [ValueSource("GetPlayerIndexes")] PlayerIndex pi, // generate tests for each combination of the PlayerIndex enum values and both float ranges
+            [Range(0.0f, 1.0f, 1.0f)] float lm,
+            [Range(0.0f, 1.0f, 1.0f)] float rm
+            )
+        {
+            Assert.IsInstanceOf<bool>(GamePad.SetVibration(pi, lm, rm));
+        }
+
+        [Test]
+        public void MaximumGamePadCount_ShouldNotReturnNull_WhenGet()
+        {
+            Assert.IsNotNull(GamePad.MaximumGamePadCount);
```

Comment:
This seems like an unnecessary test... if `GamePad.MaximumGamePadCount` returns an `int` it can never be null.